### PR TITLE
[+] Technical - Add babel

### PR DESCRIPTION
**Pull request by ArnaudValensi**
## Pull Request checklist:

- [ ] Write an explicit title for the Pull Request
- [ ] Write changes made in the CHANGELOG.md
- [ ] Test manually the implemented changes
- [ ] Review my own code (indentation, syntax, style, simplicity, readability)
- [ ] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by ArnaudValensi** [comment]
The commits are splitted to ease the review

**Comment by ArnaudValensi** [comment]
[Express - Add babel](https://trello.com/c/Yy2y62G0/2414-express-add-babel)

File: src/integrations/layer/routes.js [hunk 1]
```diff
@@ -84,11 +84,11 @@ module.exports = function (app, model, Implementation, opts) {
 
       app.get(path.generate(modelName +
         '_layer_conversations/:conversationId', opts),
-        auth.ensureAuthenticated, this.conversation);
+      auth.ensureAuthenticated, this.conversation);
 
       app.get(path.generate(modelName +
         '_layer_conversations/:conversationId/relationships/messages', opts),
-        auth.ensureAuthenticated, this.messages);
```
Comment (arnaudbesnier) [inline src/integrations/layer/routes.js]: nope

File: src/integrations/layer/routes.js [hunk 2]
```diff
@@ -84,11 +84,11 @@ module.exports = function (app, model, Implementation, opts) {
 
       app.get(path.generate(modelName +
         '_layer_conversations/:conversationId', opts),
-        auth.ensureAuthenticated, this.conversation);
```
Comment (arnaudbesnier) [inline src/integrations/layer/routes.js]: nope

File: src/integrations/mixpanel/routes.js [hunk 3]
```diff
@@ -29,6 +29,6 @@ module.exports = function (app, model, Implementation, opts) {
   this.perform = function () {
     app.get(path.generate(modelName +
       '/:recordId/relationships/mixpanel_events_this_week', opts),
-      auth.ensureAuthenticated, this.mixpanelEvents);
```
Comment (arnaudbesnier) [inline src/integrations/mixpanel/routes.js]: nope

File: dist/deserializers/params-fields.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+'use strict';
```
Comment (arnaudbesnier) [inline dist/deserializers/params-fields.js]: `dist` repository must be "git ignored".

File: .npmignore [hunk 5]
```diff
@@ -0,0 +1 @@
+src
```
Comment (arnaudbesnier) [inline .npmignore]: I think you could add `node_modules` here

Comment (ArnaudValensi) [reply .npmignore]: This is automatically done

File: dist/deserializers/params-fields.js (hunk 4 above)
Comment (ArnaudValensi) [reply dist/deserializers/params-fields.js]: If we add dist to gitignore, we then can't add the package using the git url. Both appoach are common, but if you prefer to not have an automatic commit for the transpioled code, I can do it.

Comment (arnaudbesnier) [reply dist/deserializers/params-fields.js]: Here is what I read about having an NPM package using Babel: http://2ality.com/2017/07/npm-packages-via-babel.html. Looks simple and the guy does not version the generated code.

File: package.json [hunk 6]
```diff
@@ -51,19 +52,26 @@
...
     "eslint-plugin-react": "7.5.1",
-    "git-hooks": "1.1.9",
     "gulp": "3.9.1",
     "mocha": "4.0.1",
     "moment": "2.17.1",
     "nock": "9.2.3",
     "nsp": "2.8.1",
+    "pre-commit": "1.2.2",
     "semver": "5.3.0",
+    "shelljs": "0.8.1",
```
Comment (arnaudbesnier) [inline package.json]: What is it for?

File: package.json [hunk 7]
```diff
@@ -51,19 +52,26 @@
...
     "simple-git": "1.65.0",
     "supertest": "3.0.0"
   },
   "scripts": {
+    "build": "./node_modules/babel-cli/bin/babel.js src --out-dir dist",
```
Comment (arnaudbesnier) [inline package.json]: Will we have to launch the "build" command every time we change the code in the development process?

**Pull request merged**